modules/ROOT/pages/map-users-roles-ldap-task.adoc

```diff
@@ -1,39 +1,152 @@
 
-After you perform this mapping, users in an organization can sign into Anypoint Platform using the same organizational credentials and access permissions that an organization maintains using LDAP. +
+After you perform this mapping, users in an organization can sign in to Anypoint Platform using the same organizational credentials and access permissions that an organization maintains using SAML, OpenID Connect (OIDC), or LDAP. +
 This ensures security of credentials and maintains organizational roles for accessing privileged information.
 
-Choose an external identity provider and then configure roles.
+Anypoint Platform requires different information from your identity provider based on whether you use SAML, OpenID Connect, or LDAP.
 
-To configure a role:
+== Configure Multiple Group Mappings for Single Sign-On using SAML
+
+If your organization uses single sign-on and requires its users to obtain permissions from multiple groups, you can configure a group attribute that contains all the groups you want to use. In many cases, you can create an array of groups.
+
+For example, your IDP may provide your groups in the following format:
+
+----
+<ns2:Attribute Name="Groups" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified">
+  <ns2:AttributeValue>Mule_Org_Admin_XXX</ns2:AttributeValue>
+</ns2:Attribute>
+<ns2:Attribute Name="Groups" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified">
+  <ns2:AttributeValue>Mule_Exchange_XXX</ns2:AttributeValue>
+</ns2:Attribute>
+----
+
+To prevent security vulnerabilities, Anypoint Platform requires you to create a single group that encompasses multiple group mappings:
+
+----
+<ns2:Attribute Name="Groups" NameFormat="urn:oasis:names:tc:SAML:2.0:attrname-format:unspecified">
+  <ns2:AttributeValue>Mule_Org_Admin_XXX</ns2:AttributeValue>
+  <ns2:AttributeValue>Mule_Exchange_XXX</ns2:AttributeValue>
+</ns2:Attribute>
+----
+. In Anypoint Platform, navigate to Access Management.
+. Click *Identity Providers*.
+. From *Identity Providers*, select OIDC.
+. Verify that the value in the *Group Attribute* field matches the name of the SAML attribute that contains the group names
+. In the above example of a SAML assertion, the *Group Attribute* is "Groups":
+=== Map SAML Users to Roles
 
 . In Anypoint Platform, navigate to Access Management.
 . Click *Roles*.
 . Click *Add Role* to create a role for each group of users in your organization.
-+
-image::external-identity-34af9.png[]
-
 . Specify a role name and description.
-. Click *Add Role* to add the role:
-+
-image::external-identity-c731b.png[]
+. Click *Add Role* to add the role.
+. In the *Roles* section, click the name of the new role:
+. Click *Set external group mapping*.
+. Enter the group to which you want to map.
+. Click *Set names*.
 +
+To map more than one attribute name to the selected role, click *Add More* and add another attribute.
```

Comment:
This is a step: a user action. 
Does it follow the previous step, or is it an alternative to the previous step?
Or should it actually precede the previous step?
And is "Click *Set names* a step that ends this repeated process (of more than one) or each process?